XiEditor/SidebarViewController.swift

```diff
@@ -0,0 +1,142 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+class SidebarViewController: NSViewController {
+
+    @IBOutlet weak var sidebar: NSOutlineView!
+
+    var theme: Theme {
+        return (NSApplication.shared.delegate as! AppDelegate).theme
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Setup notification observer for when the sidebar items change
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(self.updateItems),
+            name: NSNotification.Name(rawValue: "sidebarItemsChanged"),
+            object: nil
+        )
+
+        self.updateItems()
+    }
+
+    override func viewWillAppear() {
+        sidebar.backgroundColor = self.theme.background
```

Comment:
Little nit here - you don't change the text color (aka `theme.foreground`) here, which causes the sidebar to keep a dark text color even when the theme changes to a dark theme.

Reply:
@nangtrongvuon for this one I've just tested with the dark themes and the text color seems to automatically change to white/black depending on the background. Below is an example

<img width="1221" alt="screen shot 2018-07-26 at 11 06 01 am" src="https://user-images.githubusercontent.com/20056300/43235358-e43ac3d4-90c3-11e8-849b-5f92c970e444.png">

Reply:
I tried reproducing this bug, and it only happens when there's a dark theme *and* `unified_titlebar = false`. In this case, the text stays dark because [the window has an `.aqua` appearance](https://github.com/lukakerr/xi-mac/blob/sidebar/XiEditor/EditViewController.swift#L155). If you make the sidebar's appearance dark, the text will automatically change to light.

Reply:
Ah yep I can reproduce. Should be fixed now